src/shared/components/oncoprint/OncoprintUtils.ts

```diff
@@ -555,36 +648,60 @@ export function makeClinicalTracksMobxPromise(oncoprint:ResultsViewOncoprint, sa
 }
 
 export function makeHeatmapTracksMobxPromise(oncoprint:ResultsViewOncoprint, sampleMode:boolean) {
```

Comment:
why didn't you want to go the route that we did with genesets, and make it a separate category - treatmentHeatmapTracks, then have a different mobxpromise, etc? why do you want to conceptually tie gene heatmaps and treatment heatmaps?

Reply:
This is something we have considered. For us treatment profiles are just another layer of numerical properties that can be represented for sampes/patients in the oncoprint. We appreciate that treatments do not relate to genes. What we wanted to achieve, however, is a seamless mix of molecular and treament profiles that can be selected in sequence by the user and are represented in this order. This is different behavior than that of geneset tracks that are 1) selected from the query page and 2) represented always at the bottom of the heatmap. In the backend, molecular and treatment profiles are all stored in the `genetic_profile` table. All together having one mobxpromise for all profile data in the selected studies seemed like the most straight forward approach.

Reply:
To me, the main reason to keep them separate is to not separate them into different track groups - personally, I don't see a reason to tie our frontend ontology to the way the data happens to be kept in the database. I think that makes sense. 

So I think what I would want is for you to please put the functionality in `makeHeatmapTracksMobxPromise` into helper functions - one that handles genes, one that handles treatments. It's a little confusing as it is. What do you think?

Reply:
I have refactored the MobXPromisses for molecular profiles and treatment profiles. Now there are two methods for molecular and treatment heatmap tracks:
- ```function makeMolecularProfileHeatmapTracksMobxPromise(oncoprint:ResultsViewOncoprint, sampleMode:boolean) {```
and 
- ```function makeTreatmentProfileHeatmapTracksMobxPromise(oncoprint:ResultsViewOncoprint, sampleMode:boolean) {```

The `makeHeatmapTracksMobxPromise` has been refactored to use the two new partial heatmap track methods:

```
export function makeHeatmapTracksMobxPromise(oncoprint:ResultsViewOncoprint, sampleMode:boolean) {

    const moleculePromise = makeMolecularProfileHeatmapTracksMobxPromise(oncoprint, sampleMode);
    const treatmentPromise = makeTreatmentProfileHeatmapTracksMobxPromise(oncoprint, sampleMode);

    return remoteData<IHeatmapTrackSpec[]>({
        await:()=>[
            moleculePromise,
            treatmentPromise
        ],
        invoke:async()=>{
            return (moleculePromise.result!).concat(treatmentPromise.result!);
        },
        default: []
    });
}
```

Reply:
thats perfect, thanks